packages/build-tools/src/utils/processes.ts

```diff
@@ -0,0 +1,28 @@
+import spawn from '@expo/turtle-spawn';
+
+async function getChildrenPidsAsync(parentPids: number[]): Promise<number[]> {
+  const result = await spawn('pgrep', ['-P', parentPids.join(',')], {
+    stdio: 'pipe',
+  });
+  return result.stdout
+    .toString()
+    .split('\n')
+    .map((i) => Number(i.trim()))
+    .filter((i) => i);
+}
+
+export async function getAllChildrenRecursiveAsync(ppid: number): Promise<number[]> {
+  const children: number[] = [ppid];
+  let shouldRetry = true;
+  while (shouldRetry) {
+    const pids = await getChildrenPidsAsync(children);
+    shouldRetry = false;
+    for (const pid of pids) {
+      if (!children.includes(pid)) {
+        shouldRetry = true;
+        children.push(pid);
+      }
+    }
+  }
```

Comment:
Can you use the actual recursion to make this more readable?

The problem I can see in this implementation (if I'm not missing anything) is that you repeatedly call `getChildrenPidsAsync` for the same pids. You iterate the `children` list and I believe the result for the same pid can not change.